include/wil/result.h

```diff
@@ -11,6 +11,8 @@
 #ifndef __WIL_RESULT_INCLUDED
 #define __WIL_RESULT_INCLUDED
 
+#include <new> // operator new(size_t, void*)

```

Comment:
I suspect that this requires the ability to suppress (e.g. see `result_macros.h`)

Reply:
If we honored `WIL_SUPPRESS_NEW` in `<wil/result.h>`, like this:
```suggestion
#if !defined(WIL_SUPPRESS_NEW)
#include <new> // operator new(size_t, void*)
#endif
```
we would be unable to use placement `new` in `ProcessLocalStorageData<T>`, and in turn this would break `ProcessLocalStorage<T>`, which would break `g_processLocalData`. A possible alternative to `#include <new>` is reimplementing placement `new`  under `wistd`, since the implementation is trivial